drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -462,7 +462,7 @@ int sys_mm_drv_update_page_flags(void *virt, uint32_t flags)
 	tlb_entries[entry_idx] = entry;
 
 #ifdef CONFIG_MMU
-	arch_mem_map(virt, tlb_entry_to_pa(entry), CONFIG_MM_DRV_PAGE_SIZE, flags);
+	arch_mem_map(virt, va, CONFIG_MM_DRV_PAGE_SIZE, flags);
```

Comment:
sorry, don't understand, could you explain more? The second parameter of `arch_mem_map()` seems to be a physical address, whereas here you replace it with effectively a copy of the virtual address, only its cached alias?

Reply:
Physical addresses are hidden in tlb. The mmu is located between cpu and tlb. Exactly the same parameters are passed by the `sys_mm_drv_map_page` function, which works correctly.
https://github.com/zephyrproject-rtos/zephyr/blob/ba9f6279f50a982c503e1d358c9118102d5f3a06/drivers/mm/mm_drv_intel_adsp_mtl_tlb.c#L270-L272

Reply:
Ok, understand now. Can we add a description of this somewhere? I think a comment at the first of these locations would be good